handlers/session_sync_handler.go

```diff
@@ -25,19 +24,19 @@ func (h *Handler) SessionSyncHandler(w http.ResponseWriter, req *http.Request, p
 	for _, adapter := range adapters {
 		meshAdapters, _ = h.addAdapter(req.Context(), meshAdapters, prefObj, adapter.Location, provider)
 	}
-	logrus.Debugf("final list of active adapters: %+v", meshAdapters)
+	h.log.Debug("final list of active adapters: ", meshAdapters)
 	prefObj.MeshAdapters = meshAdapters
 	err := provider.RecordPreferences(req, user.UserID, prefObj)
 	if err != nil { // ignoring errors in this context
-		logrus.Errorf("unable to save session: %v", err)
+		h.log.Error(ErrSaveSession(err))
 	}
 
 	if prefObj.K8SConfig != nil && h.kubeclient != nil {
 		if prefObj.K8SConfig.ServerVersion == "" {
 			// fetching server version, if it has not already been
 			version, err := h.kubeclient.KubeClient.ServerVersion()
 			if err != nil {
-				logrus.Errorf("unable to ping the Kubernetes server")
+				h.log.Error(ErrInvalidK8SConfig)
```

Comment:
Hey @Chadha93 I think we can use a different function for the kubernetes server error.

Reply:
ErrFetchKubernetes ig

Reply:
Yeah